db/active_replicator_common.go

```diff
@@ -202,6 +203,35 @@ func (a *activeReplicatorCommon) _disconnect() error {
 	return nil
 }
 
+// stopSubChanges stops any active subChanges requests
+func (a *activeReplicatorCommon) stopSubChanges() {
+	base.TracefCtx(a.ctx, base.KeyReplicate, "closing active subChanges")
+	const (
+		maxWaitTime   = 10 * time.Second
+		retryInterval = 10 * time.Millisecond
+	)
+	ctx, cancel := context.WithDeadline(a.ctx, time.Now().Add(maxWaitTime))
```

Comment:
Using `a.ctx` here as the parent I think means it'll be cancelled from `a._stop()` before disconnect/this function is called, resulting in an immediate timeout for the retry loop instead of waiting for the deadline.

Reply:
That's probably how I saw the data race in the first place :)